variables.tf

```diff
@@ -8,6 +8,46 @@ variable "create_asg" {
   default     = true
 }
 
+variable "create_asg_with_initial_lifecycle_hook" {
+  description = "Create an ASG with intial"
+  default     = false
+}
+
+variable "initial_lifecycle_hook_name" {
+  description = "Initial lifecycle hook name required parameter"
+  default     = ""
+}
+
+variable "initial_lifecycle_hook_transition" {
+  description = "Initial Lifecycle hook transtion type"
+  default     = ""
+}
+
+variable "initial_lifecycle_hook_default_result" {
+  description = "Defines the default action the autoscaler takes when the lifecyle hook timesout"
+  default     = "CONTINUE"
+}
+
+variable "initial_lifecycle_hook_notification_metadata" {
+  description = "Metadata for the lifecycle notification event"
+  default     = ""
+}
+
+variable "initial_lifecycle_hook_heartbeat_time" {
+  description = "Lifecyle action timer"
```

Comment:
"Defines the amount of time, in seconds, that can elapse before the lifecycle hook times out. When the lifecycle hook times out, Auto Scaling performs the action defined in the DefaultResult parameter"